### DISCUSSION TITLE
Tape drives using floppy controllers

### DISCUSSION THREAD

**Discussion by vandenzel** [keirf/greaseweazle#258] [Ideas]
I like to know if someone already has used GreaseWeazle to read or write tapes using those tape drives that could be connected to a floppy controller.

I just found the specs: https://www.qic.org/html/standards/11x.x/qic117j.pdf

--
Alex.

**Comment by mike1703** [comment]
It doesn't seem like using GreaseWeazle to read or write tapes using the tape drives connected to a floppy controller is an easy task. This is because the QIC117 standard uses a different protocol over the cable than what is used with regular floppy drives.

For example, the STEP signal, which is used to switch tracks on floppy disk drives, is interpreted as a numeric command by the tape drive.
Additionally, the INDEX and TRACK ZERO signals are used completely differently by QIC tape drives, indicating readiness of the tape drive and serving as the return channel for response bits respectively.
The TRACK ZERO signal is used in floppy drives to indicate that the R/W heads are currently on track 0 of the floppy disk but it also serves as the return channel in tape drives where the tape drive pulses out one response bit at at time after the host sent a command to the tape drive. Each response bit is triggered by 2 STEP pulses, which trigger the response bits to be sent via TRACK ZERO.

These differences and additional command layers make it challenging to use GreaseWeazle with the available code for standard floppy drives from my perspective.

Nevertheless it might be interesting of the architecture of GreaseWeazle would support such a different usage of signal lines in a relatively easy way or if the floppy drive "protocol" is too much baked in.

**Comment by scratchdiver** [comment]
Did you ever get any further with this investigation? I've got a pile of these QIC tapes, unknown degradation state, and no working 486 system to read them on. putting together an old pc is not a big lift obviously but the state of flux on 35 year old tapes has me skeptical an OG floppy controller on an ancient mobo is likely to recover all the data. I don't have a strong need or the funds to pay for forensic recovery it's really just an interesting exercise possibly people can make use of in the future.
